component/espejote-templates/ipcalc.libsonnet

```diff
@@ -34,7 +34,27 @@ local parse_ip_range(prefix, rangespec) =
       end: range_parts[1],
     };
 
+local format_ipval(val) =
+  assert
+    val >= 0 && val < ipval('255.255.255.255')
+    : '%s not an IPv4 address in decimal' % val;
+
+  local iparr = std.reverse(std.foldl(
+    function(st, i)
+      local arr = st.arr;
+      local rem = st.rem;
+      {
+        arr: arr + [ rem % 256 ],
+        rem: rem / 256,
+      },
+    [ 0, 0, 0, 0 ],
+    { arr: [], rem: val }
+  ).arr);
```

Comment:
This is the functional/Jsonnet equivalent of

```rust
let mut rem = ipval;
let mut ipparts = vec![];
while rem > 0 {
  let octet = rem % 256;
  ipparts.push(octet);
  rem = rem / 256;
}
ipparts.reverse();
```

This function reverses the calculation we do in `ipval()` to get the integral value of an IPv4 address.